### DISCUSSION TITLE
how to print loss every n steps without progress bar

### DISCUSSION THREAD

**Discussion by sun-peach** [Lightning-AI/pytorch-lightning#6452] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, I am new to PL. I would like to print loss every n steps (or per epoch) without display progress bar. Now I just disable the progress bar, but I find nothing printed out, including the loss. I see my code has something like `self.log('val_loss', reduced_loss, prog_bar=False, logger=True, on_step=False, on_epoch=True, sync_dist=True)`, but cannot find where the log is. I just set `Trainer(logger=True)`. How can I get the log printed to console or some file?

Thank you very much!

**Comment by SkafteNicki** [comment]
If you can live with stuff getting logged into a file, I highly recommend that you use the `CSVLogger`:
```python
from pytorch_lightning.loggers import CSVLogger
logdir = os.getcwd() # this sets very the file will be stored
trainer = Trainer(logger=CSVLogger(logdir))
```
this will create a `metrics.csv` file in folder `default/version_0` which will be updated as training progresses.
